File: examples/c++/hello/myapp2.cpp

#include <iostream>
#include <memory>
#include "helloDSL.hpp"
#include "helloDSLSemantics.hpp"
using namespace helloDSL;
using namespace helloDSLSemantics;

int main() {
    std::shared_ptr<Start> parseTree = begin()->hello()->name("ymzk")->end();
    runHelloDSL(parseTree);
    return 0;
}

File: examples/scala/hello/myApp2.scala

import scala.language.implicitConversions

object myApp2 {
    import helloDSL._
    import helloDSLSemantics._

    def main(args: Array[String]) = {
        val parseTree: Start = begin().hello().name("ymzk").end()
        runHelloDSL(parseTree)
    }
}
